node/pkg/reporter/reporter.go

```diff
@@ -37,6 +35,7 @@ func NewReporter(ctx context.Context, opts ...ReporterOption) (*Reporter, error)
 		SubmissionInterval: groupInterval,
 		CachedWhitelist:    config.CachedWhitelist,
 		deviationThreshold: deviationThreshold,
+		JobType:            config.JobType,
```

Comment:
Let's remove this JobType and declare function property instead, I think it'll remove if branch and make the code more simple

```
type Reporter struct {
	KaiaHelper         *helper.ChainHelper
	SubmissionPairs    map[int32]SubmissionPair
	SubmissionInterval time.Duration
	CachedWhitelist    []common.Address

	contractAddress    string
	deviationThreshold float64

	nodeCtx    context.Context
	nodeCancel context.CancelFunc
	isRunning  bool

	LatestData *sync.Map // map[symbol]SubmissionData

	Job func() err
}
```